src/Compilers/VisualBasic/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.vb

```diff
@@ -0,0 +1,832 @@
+        End Sub
+
+        <Fact>
+        Public Sub TestParams()
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+Class C
+    Shared Sub M(a As Integer, b As Integer, ParamArray c() As Integer)
+    End Sub
+    Shared Sub Main()
+        M(b:=2, 3, 4)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlib45AndVBRuntime(source, parseOptions:=latestParseOptions)
+            comp.AssertTheseDiagnostics(<errors>
+BC37302: Named argument 'b' is used out-of-position but is followed by an unnamed argument
+        M(b:=2, 3, 4)
+          ~
+                                        </errors>)
+        End Sub
+
+        <Fact>
+        Public Sub TestParams2()
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+Class C
+    Shared Sub M(a As Integer, b As Integer, ParamArray c() As Integer)
+        System.Console.Write($"{a} {b} {c(0)} {c(1)} Length:{c.Length}")
+    End Sub
+    Shared Sub Main()
+        M(1, b:=2, 3, 4)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim verifier = CompileAndVerify(source, expectedOutput:="1 2 3 4 Length:2", parseOptions:=latestParseOptions)
+            verifier.VerifyDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub TestInAttribute()
+
+            Dim source =
+<compilation>
+    <file name="Program.vb"><![CDATA[
+Imports System
+
+<AttributeUsage(AttributeTargets.All, AllowMultiple:=True)>
+Public Class MyAttribute
+    Inherits Attribute
+
+    Public Dim P As Integer
+    Public Dim condition As Boolean
+
+    Public Sub New(condition As Boolean, other As Integer)
+    End Sub
+End Class
+
+<MyAttribute(condition:=true, 42)>
+<MyAttribute(condition:=true, P:=1, 42)>
+<MyAttribute(42, condition:=True)>
+Public Class C
+End Class
+    ]]></file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=latestParseOptions)
+            comp.AssertTheseDiagnostics(<errors><![CDATA[
+BC30455: Argument not specified for parameter 'condition' of 'Public Sub New(condition As Boolean, other As Integer)'.
+<MyAttribute(condition:=true, 42)>
+ ~~~~~~~~~~~
+BC30455: Argument not specified for parameter 'other' of 'Public Sub New(condition As Boolean, other As Integer)'.
+<MyAttribute(condition:=true, 42)>
+ ~~~~~~~~~~~
+BC30661: Field or property '' is not found.
+<MyAttribute(condition:=true, 42)>
+                              ~
+BC37303: Named argument expected.
+<MyAttribute(condition:=true, 42)>
+                              ~
+BC30455: Argument not specified for parameter 'condition' of 'Public Sub New(condition As Boolean, other As Integer)'.
+<MyAttribute(condition:=true, P:=1, 42)>
+ ~~~~~~~~~~~
+BC30455: Argument not specified for parameter 'other' of 'Public Sub New(condition As Boolean, other As Integer)'.
+<MyAttribute(condition:=true, P:=1, 42)>
+ ~~~~~~~~~~~
+BC30661: Field or property '' is not found.
```

Comment:
It looks like the error is reported without your change as well.

---
In reply to: [124142707](https://github.com/dotnet/roslyn/pull/20182#discussion_r124142707) [](ancestors = 124142707)